docs/guides/custom_docker_images.md

```diff
@@ -0,0 +1,111 @@
+# Custom docker images
+
+!!! Warning
+    This feature is currently in beta, and the API is likely to change. Please contact us if you are interested
+    in using this feature.
+
+If you need more customization that what cloudpickle or zip artifacts can offer, or if you just already have a pre-built
+docker image, then you can create a Model Bundle with that docker image. You will need to modify your image to run a
+web server that exposes HTTP port 5005.
+
+In our example below, we assume that you have some existing Python function `my_inference_fn` that can be imported.
+If you need to invoke some other binary (e.g. a custom C++ binary), then you can shell out to the OS to call that binary;
+subsequent versions of this document will have native examples for non-Python binaries.
+
+For choice of web server, we recommend [FastAPI](https://fastapi.tiangolo.com/lo/) due to its speed and ergonomics.
+Any web server would work, although we give examples with FastAPI. 
+
+## Step 1: Install FastAPI
+
+You can add `fastapi` to the `requirements.txt` file that gets installed as part of your Dockerfile. Alternatively,
+you can add `pip install fastapi` to the Dockerfile directly.
+
+## Step 2: Set up a web server application
+
+Inside your project workspace, create a `server.py` file with these contents:
+
+```python3
+from fastapi import FastAPI
+
+from pydantic import BaseModel
+
+import my_inference_fn  # This is your inference function, defined elsewhere within your Python codebase.
```

Comment:
This wouldn't work with doc-testing integration tests

Reply:
Hmm ok I'll just define it inline here then.